nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -408,195 +450,228 @@ def __init__(self, tensor_collectors: List[TensorCollector]) -> None:
+        return hash((self.__class__.__name__, self.inplace, self._reduction_axes, tuple(self._quantile)))
 
 
 class QuantileReducer(QuantileReducerBase):
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
         x = x[0]
-        reduction_shape = self._get_reduction_shape(x)
-        return self._tensor_processor.quantile(x, self._quantile, reduction_shape, keepdims=True)
+        axis = self._get_axis(x)
+        backend = x.backend
+        return [t for t in backend.quantile(x, self._quantile, axis=axis, keepdims=True)]
 
 
 class AbsQuantileReducer(QuantileReducerBase):
-    def __init__(
-        self,
-        reduction_shape: Optional[ReductionShape] = None,
-        quantile: Union[float, List[float]] = 0.99,
-        inplace: bool = False,
-    ):
-        super().__init__(reduction_shape, quantile, False)
-
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
-        x = self._tensor_processor.abs(x[0])
-        reduction_shape = self._get_reduction_shape(x)
-        return self._tensor_processor.quantile(x, [self._quantile], reduction_shape, keepdims=True)
+        x = x[0]
+        backend = x.backend
+        x = backend.abs(x)
+        axis = self._get_axis(x)
+        return [backend.quantile(x, self._quantile, axis=axis, keepdims=True)]
 
 
 class BatchMeanReducer(TensorReducerBase):
     def __init__(self, inplace: bool = False):
-        super().__init__(None, inplace)
+        super().__init__(channel_axis=0, inplace=inplace)
 
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
-        return [self._tensor_processor.batch_mean(x[0])]
+        x = x[0]
+        backend = x.backend
+        return [backend.mean(x, axis=0, keepdims=True)]
 
 
 class MeanPerChReducer(TensorReducerBase):
     def __init__(self, channel_dim: int = 1, inplace: bool = False):
-        super().__init__(channel_dim, inplace)
+        super().__init__(channel_axis=channel_dim, inplace=inplace)
 
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
-        return [self._tensor_processor.mean_per_channel(x[0], self._reduction_shape)]
+        x = x[0]
+        backend = x.backend
+        shape = x.shape
+        axis = self._get_axis(x)
+        retval = backend.mean(x, axis=axis, keepdims=True)
+        return [retval]
 
 
 ##################################################Aggregators##################################################
 
 
 class NoopAggregator(TensorAggregatorBase):
+    # TODO (vshampor): this seems to be a bad design since this class does not actually aggregate anything
+    #  and therefore cannot be an `Aggregator` instance. In particular this is visible in the interface violation
+    #  for the _aggregate_impl function, which is supposed to return a single tensor - the aggregate - but has to
+    #  return a list of tensors instead in order for the purpose of the class to be served.
```

Comment:
I believe overhead for empty aggregators support in a semantically correct way will be way to big

Reply:
Consistency is worth the effort, and the overhead for checking the aggregator list for a `NoopAggregator` to handle its inconsistent output locally is worse.

Reply:
As far as I understand aggregators output signature is `Any`, as aggregator object can output anything. Do you think we need to make it strict?

Reply:
The stricter the boundaries, and types, the easier it is to integrate the new code into the existing code, and to understand it limitations. `Any` is not an interface at all, you could have omitted the type hint entirely then; and if there is no definition of a more or less concrete return value type here, you cannot write any meaningful code that utilizes this default value without resorting to `isinstance` checks, everywhere where you use this function.